app/services/upstream_profile_remover.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+# Service for removing upstream profiles
+class UpstreamProfileRemover
+  class << self
+    def run!
+      policies = Profile.canonical(false)
+                        .joins(:parent_profile)
+                        .joins(:test_results)
+                        .where(parent_profiles_profiles: { upstream: true })
+                        .select(:policy_id).distinct
+
+      Policy.where(id: policies).destroy_all
```

Comment:
Why are we removing policies? AFAIK this should only remove profiles and keep other policy profiles intacted. Policy can be removed if no other profile exists, which should happen automatically (if I'm not mistaken).

Reply:
A policy is always based on a canonical profile, if the canonical profile is upstream, all the policies underneath are upstream.

Reply:
@vkrizan where can I verify this automatic deletion?

Reply:
> A policy is always based on a canonical profile, if the canonical profile is upstream, all the policies underneath are upstream.

Not necessarily. Policy is an umbrella/parent of profiles. Whether an initial profile is or isn't coming out of upstream does not necessarily make the whole policy as upstream. Users might have had created a policy at the time where the latest benchmark was the one that we now mark as upstream, however they might have had uploaded with a proper benchmark&profile that is not upstream.

Picture this scenario:
```
Policy A
  -> profile A1 (initial, upstream)
  -> profile A2 (downstream; has test results)
```

In this case it would remove everything, even-though the policy has downstream profiles with results.
If we are going to remove just the upstream ones from the scenario and the profile is initial one, then we need to find/elect a new profile or find some other means how to resolve this situation.

Policy is based off of initial profile. Canonical profile only plays role to determine "policy type", but only out of the initial profile.

> where can I verify this automatic deletion?

https://github.com/RedHatInsights/compliance-backend/blob/cea75b77678f2cc8699cf1a679337a0be94ff9d1/app/models/concerns/profile_policy_association.rb#L9